pkg/generator/github/github.go

```diff
@@ -160,8 +160,14 @@ func newGHClient(ctx context.Context, k client.Client, n string, hc *http.Client
 	if res.Spec.URL != "" {
 		gh.URL = res.Spec.URL + ghPath
 	}
+
+	ns := n
+	if res.Spec.Auth.PrivateKey.SecretRef.Namespace != nil && *res.Spec.Auth.PrivateKey.SecretRef.Namespace != "" {
```

Comment:
The ClusterGenerator only creates Generators in the namespace they are required in. This is a violation of that principle where it could access / try accessing a secret in a different namespace to the one in which the generator is created in.

This is not a desired behaviour.